ulog/log.go

```diff
@@ -186,6 +193,16 @@ func (l *baseLogger) fieldsConversion(keyVals ...interface{}) []zap.Field {
 				fields = append(fields, zap.Marshaler(key, value))
 			case fmt.Stringer:
 				fields = append(fields, zap.Stringer(key, value))
+			case stackTracer:
+				stack := value.StackTrace()
+				lines := make([]zap.Field, 0, len(stack))
+				for _, frame := range stack {
+					s := fmt.Sprintf("%s", frame)
+					l := fmt.Sprintf("line:%d", frame)
```

Comment:
I don't think zap supports arrays: only key/value pairs. I am mostly concerned about the size of the logged string, the entire trace will be printed in one line. Logging each line separately should be better, but I am afraid we can lose error then.(In terms of context, something else can be logged at the same time)